homeassistant/components/recorder/purge.py

```diff
@@ -46,6 +37,13 @@ def purge_old_data(instance, purge_days, repack):
         # Otherwise, if the SQL server has "ON DELETE CASCADE" as default, it
         # will delete the protected state when deleting its associated
         # event. Also, we would be producing NULLed foreign keys otherwise.
+        protected_events = session.query(States.event_id) \
+            .filter(States.state_id.in_(protected_state_ids)) \
```

Comment:
I got a warning from pytest when running the purge tests locally. I think that it is from this line.

```
tests/components/recorder/test_purge.py::TestRecorderPurge::test_purge_old_events
  /Users/paulus/dev/python/py34-home-assistant/lib/python3.4/site-packages/sqlalchemy/sql/default_comparator.py:161: SAWarning: The IN-predicate on "states.state_id" was invoked with an empty sequence. This results in a contradiction, which nonetheless can be expensive to evaluate.  Consider alternative strategies for improved performance.
    'strategies for improved performance.' % expr)
  /Users/paulus/dev/python/py34-home-assistant/lib/python3.4/site-packages/sqlalchemy/sql/default_comparator.py:161: SAWarning: The IN-predicate on "events.event_id" was invoked with an empty sequence. This results in a contradiction, which nonetheless can be expensive to evaluate.  Consider alternative strategies for improved performance.
    'strategies for improved performance.' % expr)
```